iamclient/iamclient.go

```diff
@@ -51,32 +47,56 @@ const (
 	iamReconnectInterval = 10 * time.Second
 )
 
+const iamCertType = "iam"
+
 /***********************************************************************************************************************
  * Types
  **********************************************************************************************************************/
 
 // Client IAM client instance.
 type Client struct {
 	sync.Mutex
+	*grpchelpers.IAMPublicServiceClient
 
-	sender Sender
+	config        *config.Config
+	sender        Sender
+	cryptocontext *cryptutils.CryptoContext
+	insecure      bool
 
 	nodeID   string
 	systemID string
 
-	publicConnection    *grpc.ClientConn
+	publicConnection    *grpchelpers.BlockingGRPCConn
 	protectedConnection *grpc.ClientConn
-	publicService       pb.IAMPublicServiceClient
+
 	identService        pb.IAMPublicIdentityServiceClient
 	certificateService  pb.IAMCertificateServiceClient
 	provisioningService pb.IAMProvisioningServiceClient
+	publicNodesService  pb.IAMPublicNodesServiceClient
+	nodesService        pb.IAMNodesServiceClient
+
+	nodeInfoSubs *nodeInfoChangeSub
+
+	tlsCertChan      <-chan *pb.CertInfo
+	closeChannel     chan struct{}
+	disableReconnect atomic.Bool
```

Comment:
Use ordinal bool and guard with mutex. It is already guarded in most the cases.

Reply:
it is required to skip reconnecting when it is in process. atomic fits better for this particular case